main.tf

```
@@ -40,6 +40,23 @@ resource "azurerm_storage_account" "this" {
   }
 }
 
+resource "azurerm_monitor_diagnostic_setting" "this" {
+  name                       = "logs-${var.application}-${var.environment}"
```

Diagnostic setting is only created for this storage account, not the entire environment. The name of the diagnostic setting should reflect this:

```suggestion
  name                       = "${azurerm_storage_account.this.name}-logs"
```